main.py

```diff
@@ -120,20 +149,64 @@ def update_board(board):
+    c = config.Size * config.Width + config.Edge / 2 + config.ButtonSize / 2  # The middle of the line of buttons
+    d = config.StartOfSlider
+    e = config.EndOfSlider
+    if y < d:
+        y = d
+    elif y > e:
+        y = e
+    pygame.draw.rect(Screen, config.Background, ((c + config.NotchLength / 2, d - config.NotchLength),
+                                                 (config.Width * config.Size + config.Edge + config.Border +
+                                                  config.ButtonSize, e)))
+    pygame.draw.line(Screen, (180, 180, 180), (c, d), (c, e))
+    f = (e - d) / config.Notches  # How far each notch goes
+    for g in range(config.Notches + 1):
+        pygame.draw.line(Screen, (180, 180, 180), (c - config.NotchLength / 2, d + g * f),
+                         (c + config.NotchLength / 2, d + g * f))
+    config.write(Screen, c - (12 + config.NotchLength), (d + e) * 0.5, "Speed", (180, 180, 180), 20,
+                 rotate=90, alignment=("left", "centre"))
+    if fps_limit:
+        colour = (0, 255, 100)
+    else:
+        colour = (160, 160, 160)
+    pygame.draw.polygon(Screen, colour, ((c + config.NotchLength / 2, y),
+                                         (c + config.NotchLength, y - config.NotchLength / 2),
+                                         (c + 2 * config.NotchLength, y - config.NotchLength / 2),
+                                         (c + 2 * config.NotchLength, y + config.NotchLength / 2),
+                                         (c + config.NotchLength, y + config.NotchLength / 2)))
+
+
 pygame.init()
-Screen = pygame.display.set_mode((config.Size * config.Width, config.Size * config.Height))
+Screen = pygame.display.set_mode((config.Size * config.Width + config.ButtonSize, config.Size * config.Height))
 Screen.fill(config.Background)
-pygame.display.set_caption("Game of Life")
 Board = clean_board()
 Paused = True
+FPS = 10
+FPSLimit = True
+draw_fps_slider(((maths.log(FPS, 10) + 1) / -3) * (config.EndOfSlider - config.StartOfSlider) + config.EndOfSlider,
+                FPSLimit)
+LastFrame = time.time()  # The time when the last frame update happened.
+Generations = 0
+
+for _ in range(int(config.Width * config.Height / 5)):
+    rx = random.randint(config.Cushion, config.Cushion + config.Width - 1)
+    ry = random.randint(config.Cushion, config.Cushion + config.Height - 1)
+    Board[rx][ry].CurrentState = 0
+    Board[rx][ry].birth(config.Square)  # random.randint(0, config.NoOfButtons - 1))
 
 while True:
-    Board, Paused, OneTurn = check_user_input(Board, Paused)  # If game is paused OneTurn allows you
-    Board = update_board(Board)                                 # to go forward one turn at a time
-    if not Paused or (Paused and OneTurn):
+    Board, Paused, OneTurn, FPS, FPSLimit = check_user_input(Board, Paused, FPS, FPSLimit)
+    Board = update_board(Board)
+    if (not Paused or (Paused and OneTurn)) and ((not FPSLimit) or time.time() - LastFrame > 1 / FPS):
```

Comment:
Why are you controlling the fps yourself? You can do 
```
clock = pygame.time.Clock()
clock.tick(60)
```
https://stackoverflow.com/questions/35617246/setting-a-fixed-fps-in-pygame-python-3

Reply:
I tried that but it pauses the game at that point, like the time.sleep() function. This means that an input cannot be obtained between frames